src/Common/src/System/Memory/PinnedSpanMemoryManager.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers
+{
+    internal unsafe class PinnedSpanMemoryManager<T> : MemoryManager<T> where T : struct
+    {
+        private readonly void* _pointer;
+        private readonly int _length;
+
+        internal PinnedSpanMemoryManager(void* pointer, int length)
+        {
+            _pointer = pointer;
+            _length = length;
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+        }
+
+        public override Span<T> GetSpan()
+        {
+            return new Span<T>(_pointer, _length);
+        }
+
+        public override MemoryHandle Pin(int elementIndex = 0)
+        {
+            throw new NotImplementedException();
```

Comment:
Why NotImplementedException rather than `return default;`?  Or if we want to throw an exception because it'll never be used, how about NotSupportedException?  NotImplementedException suggests we just haven't gotten around to it yet.